mypyc/irbuild/constant_fold.py

```diff
@@ -72,6 +75,35 @@ def constant_fold_expr(builder: IRBuilder, expr: Expression) -> ConstantValue |
         value = constant_fold_expr(builder, expr.expr)
         if value is not None and not isinstance(value, bytes):
             return constant_fold_unary_op(expr.op, value)
+    # we can also constant fold some common methods of builtin types
+    elif isinstance(expr, CallExpr) and isinstance(callee := expr.callee, MemberExpr):
+        folded_callee = constant_fold_expr(builder, callee.expr)
+
+        # builtins.str methods
```

Comment:
We can constant fold quite a lot of cases for builtins without too much code. I have an idea for how to implement this cleanly but that's outside the scope of this PR.